pyro/contrib/glmm/glmm.py

```diff
@@ -358,7 +368,22 @@ def analytic_posterior_cov(prior_cov, x, obs_sd):
     return posterior_cov
 
 
-def iter_plates_to_shape(shape):
-    # Go backwards (right to left)
-    for i, s in enumerate(shape[::-1]):
-        yield pyro.plate("plate_" + str(i), s)
+def _broadcast_shape(shapes):
+    r"""
+    Given a list of tensor sizes, returns the size of the resulting broadcasted
+    tensor.
+    Args:
+        shapes (list of torch.Size): list of tensor sizes
+    Copied from PyTorch 0.4.0
+    """
+    shape = torch.Size()
+    for s in shapes:
+        shape = torch._C._infer_size(s, shape)
+    return shape
```

Comment:
There is a `torch.broadcast_tensors(*tensors)` which will directly broadcast all the tensors, so it will likely need some code modification. I remember @fritzo made a change to torch distributions to replace `torch._C._infer_size` with `torch.broadcast_tensors` for compatibility with JIT. Unless that is fixed, this will likely create issues with JIT.

Reply:
I'll rewrite to use `torch.broadcast_tensors`- this code was probably for compatibility with a now-very-outdated version to torch